### ISSUE TITLE
Ensure that user cannot leave dialog while it is open

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/balloons-and-static-electricity#123]
@terracoda mentioned: 

> Dialog issue: the user ended up back in the content somehow without closing the dialog.

I will need to thoroughly test the keyboard help dialog to understand how the user was able to leave the dialog without hitting 'escape` or the close button.

**Comment by jessegreenberg** [comment]
From #124, @terracoda mentioned that the user left dialog content without closing the dialog after using the ARROW keys.  She also provided useful resources that could solve this issue (Accessibility Questions Google Doc).

Very helpful information, thanks @terracoda!

**Comment by jessegreenberg** [comment]
This is not something I am able to reproduce with any screen reader or platform.  We have been using this source for accessible dialogs:
https://accessibility.oit.ncsu.edu/training/aria/modal-window/version-3/

The most important items from this are:
* When modal dialog is displayed an overlay is placed over the main pate so it is not clickable and visually grayed.
* When dialog is open, the main page is hidden with ```aria-hidden=true``` to prevent screen readers from accessing content under the dialog
* Title of the dialog is identified with aria-labelledby
* Beginning of dialog is marked with an H1
* There are off screen instructions used to describe dialog and how to interact with it
* Document role should be used
  * NOTE: Documentation suggests that this is not needed anymore because of changes to the way NVDA browses content. This is contrary to our testing and results.

Most importantly, the div containing the screen view and all of its elements has the ```hidden``` attribute when the dialog is open.  According to some documentation from The Paciello Group (https://www.paciellogroup.com/blog/2012/05/html5-accessibility-chops-hidden-and-aria-hidden/), user agents should *not* render this content in the accessibility tree.

**Comment by jessegreenberg** [comment]
According to the Paciello Group (https://www.paciellogroup.com/blog/2016/01/the-state-of-hidden-content-support-in-2016/) :

* All major screen readers and browsers support ```aria-hidden``` to hide visible content from a screen reader
* All major screen readers and browsers support ```hidden``` to hide content from both AT and View.

See http://stevefaulkner.github.io/HTML5accessibility/tests/hidden-2016.html for a *very* comprehensive list (thanks Paciello group!) of platform testing.

Important note from above matrix: For some browsers, if both ```aria-hidden``` and ```hidden``` are used, the content is actually **unhidden**!

Important note from documentation notes: Edge is not being tested because it does not have sufficient accessibility support to be useful in testing.  I have had similar experiences with Edge.

**Comment by jessegreenberg** [comment]
For this case, I think ```aria-hidden``` is more appropriate for hiding the screen view.  The ScreenView is still visible, though should be hidden from AT.   I will change this in BASE and verify.

**Comment by jessegreenberg** [comment]
```aria-hidden``` does not fully remove elements from navigation in the same way ```hidden``` does.  Virtual cursor is limited to the dialog, but if user tabs out of dialog and back to the address bar, they can find the elements from there.  Going back to https://accessibility.oit.ncsu.edu/training/aria/modal-window/version-3/ we need to more aggressively control the tab navigation.

**Comment by jessegreenberg** [comment]
The keyboard help dialog also receives focus.  I don't think it should.  The 'close' button is the only element that should receive focus in this case.  We can prevent default ```tab``` behavior on the button, and restrict focus to this window.

As in the example provided, this means that the user cannot get to the browser address bar if the pop up dialog is open.  Perhaps this is acceptable as long as the Dialog is easy to use, and it can be closed in obvious ways.

**Comment by jessegreenberg** [comment]
If we restrict focus to a single button, or force the focus in any way, some screen readers automatically transition into forms mode.  I notice that in the example dialog above, NVDA must be in forms mode, the virtual cursor cannot be used.  We can try adding the ```document``` role to the close button, this may override this. This adds some complexity, I am hesitant to overload with aria attributes.

**Comment by jessegreenberg** [comment]
Adding the document role to a button does not help.  Perhaps if make the section in the popup dialog focusable?  This way, the user can navigate to the content and the close button.  The docuemtn role on the text should allow the user to navigate that content with the arrow keys and then close the dialog with the button.  This would also leave the dialog window out of focus, which is a good thing.

**Comment by jessegreenberg** [comment]
We need a visual representation of the keyboard help content.  I will pull something together with text very quicly.  This content should receive focus as a document.  We can work on styling later in another issue.  This way, focus is restricted to content and the close button.

**Comment by jessegreenberg** [comment]
The keyboard help dialog now 

 - Uses JavaScript to restrict navigation to the tab navigation to the content and the close button
 - Uses ```aria-hidden``` on the screen view container to hide all other simulation components from the virtual cursor
 - Uses a document role so that content can be navigated with the arrow keys
 - Does *not* use the ```dialog``` role, it seems to override the ```document``` role behavior.
 - Looks like this (for now):
![help](https://cloud.githubusercontent.com/assets/6396244/12937228/b75d30b4-cf63-11e5-8356-8c0ebd495515.PNG)

**Comment by jessegreenberg** [comment]
Commit 679ba5817df5610fd77d62364b9b57fa1ad1b0ed handles this.

@terracoda, I think this issue should be fixed.  Between all of the items in https://github.com/phetsims/balloons-and-static-electricity/issues/123#issuecomment-182172294, I think navigation should be restricted to the dialog.  If you have any comments or encounter this issue again, please feel free to reopen this issue and let me know.
